add static dispatch kernel registration to open source

dolpm opened the pull request:
Summary: static dispatch registry should be moved to open source. the rest can maintain internally for now, since delegates will all go through ET hop.

Test Plan: spot checked existing tests and didn't see any missing registrations

Differential Revision: D80099377

SherlockNoMad reviewed (approved):
register_static_cpu_kernel_handlers plz

dolpm committed:
add static dispatch kernel registration to open source (#160439)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/160439

static dispatch registry should be moved to open source. the rest can maintain internally for now, since delegates will all go through ET hop.

Test Plan: spot checked existing tests and didn't see any missing registrations

Reviewed By: zhxchen17, SherlockNoMad

Differential Revision: D80099377
